### DISCUSSION TITLE
Descending order in queryOrdered(byChild:) function

### DISCUSSION THREAD

**Discussion by onl1ner** [firebase/firebase-ios-sdk#8142] [Q&A]
Hello, I am wondering why there is no way to get data sorted in descending order? For example, if there is a need to retrieve information sorted by date to show newest one at the beginning.

I thought about reversing the resulting array, but I have data paging so it is not a case for me. I stopped at the idea to make my timestamps negative and format them at the client, but this is so hacky.

Thanks!

**Comment by unixneo** [comment]
Hey @onl1ner 

I am also perplexed on why sorting on a `TimeInterval` across the results of `FirestoreQuery()` return data is nearly impossible, or a best very difficult.  Have spent hours and hours on this "simple 10 second" out-of-the-box task in Ruby or Python (for example).

Sorting on DB queries is a basic core function, but  in this SwiftUI extension for Firebase, something as basic as:

```swift
@FirestoreQuery var items: [ToDoListItem]. //  struct for todo list item
```

Trying to sort on a simple `TimeInterval`  `var` in a `struct` seems to be forgotten in the `Firebase` ecosystem.

See, for example:

https://developer.apple.com/forums/thread/737760

It's odd to travel into SwiftUI and Firebase from a Ruby and Python background only to find sorting on a basic query return cannot be done easily and simply.   Hours of Google'n turned up nothing workable.

😞

**Comment by peterfriese** [comment]
Hi @unixneo -

sorting in ascending / descending order should be possible:

Using @FirebaseQuery
```
  @FirestoreQuery(
    collectionPath: "todos",
    predicates: [
      .where("isDone", isEqualTo: false),
      .order(by: "dateAdded", descending: true)
    ]
  ) fileprivate var todos: Result<[Todo], Error>
```

Using a query directly (as mentioned by @onl1ner ):
```
  let query = db.collection("todos")
    .whereField("userId", isEqualTo: self.userId)
    .order(by: "dateAdded", descending: true)
```

LMK if this helps.
